rpc/client.go

```diff
@@ -209,9 +209,9 @@ func initClient(conn ServerCodec, idgen func() ID, services *serviceRegistry) *C
 		isHTTP:      isHTTP,
 		services:    services,
 		writeConn:   conn,
-		close:       make(chan struct{}),
-		closing:     make(chan struct{}),
-		didClose:    make(chan struct{}),
+		close:       make(chan struct{}, 1),
```

Comment:
There is a send on `close` in the `Client.Close` method, but the send depends on the channel being not buffered. The expected operation in the `Client.Close` method is: if the main loop of the client is running, it will accept the send. Otherwise, the receive from the `didClose` channel will succeed instead. Adding the buffer breaks this because the send on the `close` channel will succeed even if the loop is not running.